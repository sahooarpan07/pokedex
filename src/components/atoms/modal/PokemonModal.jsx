import {Modal} from 'react-native';
import React from 'react';

const PokemonModal = ({children, visible, onRequestClose}) => {
  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={visible}
      onRequestClose={onRequestClose}>
      {children}
    </Modal>
  );
};

export default PokemonModal;
