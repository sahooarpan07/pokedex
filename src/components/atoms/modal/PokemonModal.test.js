/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonModal from './PokemonModal';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render Pokemon Modal', () => {
  const tree = renderer
    .create(<PokemonModal visible={true} onRequestClose={jest.fn()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
