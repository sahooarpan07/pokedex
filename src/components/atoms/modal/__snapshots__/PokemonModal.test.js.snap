// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pokemon Modal 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
/>
`;
